src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentCompletionEndpointTest.cs

```diff
@@ -313,6 +313,59 @@ The end.
              snippetLabels: ["snippet1", "snippet2"]);
     }
 
+    [Fact]
+    public async Task HtmlSnippetsCompletion_EmptyDocument()
+    {
+        await VerifyCompletionListAsync(
+            input: """
+                $$
+                """,
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Explicit,
+                 TriggerCharacter = null,
+                 TriggerKind = RoslynCompletionTriggerKind.Invoked
+             },
+             expectedItemLabels: ["snippet1", "snippet2"],
```

Comment:
There is an unfortunate behavior change/regression in empty HTML document in HTML LSP editor. It no longer shows tags (as non-LSP editor and thus legacy Razor editor used to). E.g. legacy HTML

<img width="236" alt="image" src="https://github.com/user-attachments/assets/36eb018c-65c0-43f4-ace3-54d1eba81de5" />

but no completion at all in LSP HTML editor (since there aren't any snippets). By extension that affects Razor as well.

Reply:
No issues with changes in this PR though, thank you for fixing this case!

Reply:
Same conversation as https://github.com/dotnet/razor/pull/11343#discussion_r1900495785 really.

This PR isn't introducing any regression or change in behaviour. We don't show Html elements or components when completion is invoked on an empty line, as the completion items don't have the `<` in them, so they wouldn't produce valid code. This PR just makes the experience the same whether the document is empty or not.